pythran/dist.py

```diff
@@ -7,44 +7,40 @@
 
 import os.path
 import os
+from warnings import warn
+
+from distutils.command.build_ext import build_ext as LegacyBuildExt
 
 from numpy.distutils.extension import Extension
 
-from numpy.distutils.command import build_ext as build_ext0
-from distutils.command import build_ext as build_ext1
-
-for build_ext in (build_ext0, build_ext1):
-
-    LegacyBuildExt = build_ext.build_ext
-
-    class PythranBuildExt(LegacyBuildExt):
-
-        # workaround mismatch between distutils and numpy
-        # implementation
-        def swig_sources(self, sources, *args):
-            return sources
-
-        def build_extension(self, ext):
-            prev = {'preprocessor': None,
-                    'compiler_cxx': None,
-                    'compiler_so': None,
-                    'compiler': None,
-                    'linker_exe': None,
-                    'linker_so': None}
-            if getattr(ext, 'cxx', None) and 'CXX' not in os.environ:
-                for attr in prev.keys():
-                    prev[attr] = getattr(self.compiler, attr)[0]
-                    getattr(self.compiler, attr)[0] = ext.cxx
-
-                try:
-                    return LegacyBuildExt.build_extension(self, ext)
-                finally:
-                    for attr in prev.keys():
-                        getattr(self.compiler, attr)[0] = prev[attr]
-            else:
-                return LegacyBuildExt.build_extension(self, ext)
 
-    build_ext.build_ext = PythranBuildExt
+class PythranBuildExt(LegacyBuildExt):
+    """Subclass of `distutils.command.build_ext.build_ext` which is required to build
+    `PythranExtension` with the configured C++ compiler. It may also be
+    subclassed if you want to combine with another build_ext class (NumPy,
+    Cython implementations).
+
+    """
+    def build_extension(self, ext):
+        prev = {'preprocessor': None,
+                'compiler_cxx': None,
+                'compiler_so': None,
+                'compiler': None,
+                'linker_exe': None,
+                'linker_so': None}
+
+        if hasattr(ext, 'cxx'):
```

Comment:
I know it's my original code, but maybe a comment here to state we're saving the environment?